src/uct/api/v2/uct_v2.h

```diff
@@ -251,11 +251,16 @@ typedef enum {
 
 typedef enum {
     /**
-     * Indicates that remote access to the returned remote key should fail if
-     * @ref uct_md_mem_dereg_v2 is called with the flag
-     * @ref UCT_MD_MEM_DEREG_FLAG_INVALIDATE on the memory handle. If this flag
-     * is not set, using the remote key after invalidating the memory region has
-     * an undefined behavior.
+     * The flag is used indicate that remote access to a memory region
+     * associated with the remote key must fail once the memory region is
+     * deregister using @ref uct_md_mem_dereg_v2 and
+     * @ref UCT_MD_MEM_DEREG_FLAG_INVALIDATE flag set. Using
+     * @ref uct_md_mem_dereg_v2 deregistration routine with
+     * @ref UCT_MD_MEM_DEREG_FLAG_INVALIDATE flag set while the rkey was
+     * generated without @ref UCT_MD_MKEY_PACK_FLAG_INVALIDATE flag set may
+     * result in data corruption. In other words in order for
```

Comment:
@yosefe I thought access to outdate translation table will result in silent data corruption. I don't like "undefined behavior" because one the interpretation of this that your code will be "fine", which is dangerous in this case.

Reply:
maybe 
```
     * @ref UCT_MD_MEM_DEREG_FLAG_INVALIDATE flag set. Using
     * @ref uct_md_mem_dereg_v2 deregistration routine with
     * @ref UCT_MD_MEM_DEREG_FLAG_INVALIDATE flag set on an rkey that was
     * generated without @ref UCT_MD_MKEY_PACK_FLAG_INVALIDATE flag will
     * not function correctly, and may result in data corruption.
```

Reply:
Agreed - I think it is better. 

Can we cache the UCT_MD_MKEY_PACK_FLAG_INVALIDATE status on memh and assert if UCT_MD_MEM_DEREG_FLAG_INVALIDATE is set but UCT_MD_MKEY_PACK_FLAG_INVALIDATE  was not used during pack ? I guess there are some cases when the rkey is never packet but you can capture all states and raise the assert only if key was packet.

Reply:
yes, @evgeny-leksikov will add it as part of the implementation of IB (only MD that supports invalidation for now).
other MDs do not support invalidation and will fail with UNSUPPORTED anwyay.